File: app/src/main/java/com/suhang/networkmvp/dagger/component/BaseComponent.java
package com.suhang.networkmvp.dagger.component;


import com.suhang.networkmvp.annotation.BaseScope;
import com.suhang.networkmvp.dagger.module.BaseModule;
import com.suhang.networkmvp.dagger.module.BlankModule;

import dagger.Subcomponent;

/**
 * Created by 苏杭 on 2017/1/20 16:26.
 */

@BaseScope
@Subcomponent(modules = BaseModule.class)
public interface BaseComponent {
    BlankComponent providerBlankComponent(BlankModule blankModule);
}

File: app/src/main/java/com/suhang/networkmvp/dagger/component/BaseComponent.kt
package com.suhang.networkmvp.dagger.component
import com.suhang.networkmvp.annotation.BaseScope
import com.suhang.networkmvp.dagger.module.BaseModule
import com.suhang.networkmvp.dagger.module.BlankModule
import com.suhang.networkmvp.dagger.module.ChildModule
import com.suhang.networkmvp.dagger.module.HomeModule
import dagger.Subcomponent

@BaseScope
@Subcomponent(modules = arrayOf(BaseModule::class))
interface BaseComponent {
    fun providerBlankComponent(blankModule: BlankModule): BlankComponent
    fun providerHomeComponent(homeModule: HomeModule): HomeComponent
    fun providerAttentionComponent(attentionModule: ChildModule): ChildComponent
}